src/schemas.jl

```diff
@@ -530,11 +554,13 @@ function _generate_record_type_definitions(schema_version::SchemaVersion, record
                 $(field_assignments...)
                 return new{$(type_param_names...)}($(keys(record_fields)...))
             end
-        end
-        outer_constructor_definitions = quote
             function $R(; $(field_kwargs...))
-                return $R{$((:(typeof($n)) for n in names_of_parameterized_fields)...)}(; $(keys(record_fields)...))
+                $parent_record_application
+                $(field_assignments...)
+                return new{$((:(typeof($n)) for n in names_of_parameterized_fields)...)}($(keys(record_fields)...))
```

Comment:
why are we switching from kwargs to positional here?

Reply:
oh I see, because we're not falling back to calling `$R` constructor here anymore, makes sense.

Reply:
There's some real subtle stuff going on here. In #82 I changed this code to no longer be an inner constructor and just make the non-parametric constructor call the parametric constructor. However, with the changes made in this PR it turned out that if a user passed in a value that wasn't a subtype of a specified parametric type we would no longer see the nice error messages and get a very confusing `TypeError` that included the `gensym`d parametric variable which is normally hidden from end users.

I also made this PR https://github.com/beacon-biosignals/Legolas.jl/pull/84 which also makes this change and introduces additional tests specifically to look for this

Reply:
Now that #82 is merged and this PR has merged in those changes these changes no longer exist here :)